lib/bundler.rb

```diff
@@ -195,6 +195,16 @@ def tmp(name = Process.pid.to_s)
 
     def rm_rf(path)
       FileUtils.remove_entry_secure(path) if path && File.exist?(path)
+    rescue ArgumentError => e
+      if /world writeable/ =~ e.message
```

Comment:
why not check the permission bits on the parent directory before trying to remove it instead of rescuing the exception?